packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+    executionSettings,
+    checkSignature = false,
+  }: {
+    account: Address;
+    checkSignature?: boolean;
+    walletClient: WalletClient;
+    collection: PremintResponse["collection"];
+    token: MintArgumentsSettings;
+    publicClient?: PublicClient;
+    executionSettings?: {
+      deleted?: boolean;
+      uid?: number;
+    };
+  }) {
+    publicClient = this.getPublicClient(publicClient);
+
+    const newContractAddress = await publicClient.readContract({
+      address: this.getExecutorAddress(),
+      abi: zoraCreator1155PremintExecutorImplABI,
+      functionName: "getContractAddress",
+      args: [collection],
+    });
+
+    const tokenConfig = {
+      ...DefaultMintArguments,
+      fixedPriceMinter: this.getFixedPriceMinterAddress(),
+      royaltyRecipient: account,
+      ...token,
+    };
+
+    let uid = executionSettings?.uid;
+    if (!uid) {
+      const uidResponse = await this.get(
+        `${ZORA_API_BASE}signature/${
+          this.network.zoraBackendChainName
+        }/${newContractAddress.toLowerCase()}/next_uid`,
+      );
+      uid = uidResponse["next_uid"];
+    }
+
+    if (!uid) {
+      throw new Error("UID is missing but required");
+    }
+
+    let deleted = executionSettings?.deleted || false;
+
+    const premintConfig = {
+      tokenConfig: tokenConfig,
+      uid,
+      version: 1,
+      deleted,
+    };
+
+    return await this.signAndSubmitPremint({
+      uid,
+      verifyingContract: newContractAddress,
+      premintConfig,
+      checkSignature,
+      publicClient,
+      walletClient,
+      collection,
+    });
+  }
+
+  /**
+   * Fetches given premint data from the ZORA API.
+   *
+   * @param address Address for the premint contract
+   * @param uid UID for the desired premint
+   * @returns PremintResponse of premint data from the API
+   */
+  async getPremintData(address: string, uid: number): Promise<PremintResponse> {
+    const response = await this.get(
+      `${ZORA_PREMINT_API_BASE}signature/${this.network.zoraBackendChainName}/${address}/${uid}`,
+    );
+    return response as PremintResponse;
+  }
+
+  /**
+   * Check user signature for v1
+   *
+   * @param data Signature data from the API
+   * @returns isValid = signature is valid or not, contractAddress = assumed contract address, recoveredSigner = signer from contract
+   */
+  async isValidSignature({
+    data,
+    publicClient,
+  }: {
+    data: PremintResponse;
+    publicClient?: PublicClient;
+  }): Promise<{
+    isValid: boolean;
+    contractAddress: Address;
+    recoveredSigner: Address;
+  }> {
+    publicClient = this.getPublicClient(publicClient);
+
+    const [isValid, contractAddress, recoveredSigner] =
+      await publicClient.readContract({
+        abi: zoraCreator1155PremintExecutorImplABI,
+        address: this.getExecutorAddress(),
+        functionName: "isValidSignature",
+        args: [data.collection, convertPremint(data.premint), data.signature],
+      });
```

Comment:
Do we need to `try/catch` this?

Reply:
If it fails, the error should just bubble up. It shouldn't fail unless there is an RPC call.